.gitignore

```diff
@@ -35,3 +35,5 @@ out/
 
 ### VS Code ###
 .vscode/
+
+/src/main/resources/application.properties
```

Comment:
Colocar o application.properties no gitignore não é uma boa ideia.

Você pode usar variaveis de ambiente para esconder as informações confidenciais:
`uri: ${MONGO_DATABASE_URI:mongodb://localhost:27017}`

Na configuração acima a uri sera substituida por uma variavel de ambiente se ela existir. Se não existir vai usar a uri que esta depois dos dois pontos

Reply:
Um pouco mais sobre esse tipo de configuração
https://blog.devgenius.io/working-with-application-properties-in-spring-boot-79e72630bba0